src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -650,6 +650,25 @@ private TypeElement getElement(Class<?> clazz) {
         + "}\n");
   }
 
+  @Test public void annotation() throws Exception {
+    TypeSpec annotation = TypeSpec.annotationBuilder("MyAnnotation")
+        .addModifiers(Modifier.PUBLIC)
+        .addMethod(MethodSpec.methodBuilder("test")
+                .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
+                .initializer("$L", 0)
+                .returns(int.class)
+                .build())
+        .build();
+
+    assertThat(toString(annotation)).isEqualTo(""
+        + "package com.squareup.tacos;\n"
+        + "\n"
+        + "public @interface MyAnnotation {\n"
+        + "  int test() default 0;\n"
```

Comment:
To validate the API, could you add a test that generates a default value being an annotation?

```
@interface Foo {
  int value();
}
@interface Bar
  Foo value() default @Foo(0)
}
```

(it might be something we need to test when emitting annotations too: are we able to emit `@Bar(@Foo(42)) int someField`?)